packages/@aws-cdk/aws-codedeploy/lib/lambda/deployment-group.ts

```diff
@@ -0,0 +1,196 @@
+          .addServicePrincipal('codedeploy.amazonaws.com'));
+      }
+    } else {
+      serviceRole = new iam.Role(this, 'ServiceRole', {
+        assumedBy: new iam.ServicePrincipal('codedeploy.amazonaws.com')
+      });
+    }
+    serviceRole.addToPolicy(new iam.PolicyStatement()
+      .addResource('arn:aws:s3:::*/CodeDeploy/*')
+      .addActions('s3:GetObject', 's3:GetObjectVersion'));
+    serviceRole.addToPolicy(new iam.PolicyStatement()
+      .addResource('*')
+      .addCondition('StringEquals', {
+        's3:ExistingObjectTag/UseWithCodeDeploy': 'true'
+      })
+      .addActions('s3:GetObject', 's3:GetObjectVersion'));
+    serviceRole.addToPolicy(new iam.PolicyStatement()
+      .addResource(props.alias.functionArn)
+      .addActions('lambda:UpdateAlias', 'lambda:GetAlias'));
+    if (props.alarms) {
+      serviceRole.addToPolicy(new iam.PolicyStatement()
+        .addResources(...props.alarms.map(alarm => alarm.alarmArn))
+        .addAction('cloudwatch:DescribeAlarms'));
+    }
+
+    const alarms = props.alarms || [];
+
+    const resource = new CfnDeploymentGroup(this, 'Resource', {
+      applicationName: props.application.applicationName,
+      serviceRoleArn: serviceRole.roleArn,
+      deploymentGroupName: props.deploymentGroupName,
+      deploymentConfigName: `CodeDeployDefault.Lambda${props.deploymentConfig}`,
+      deploymentStyle: {
+        deploymentType: props.deploymentConfig === LambdaDeploymentConfig.AllAtOnce ?
+          DeploymentType.InPlace : DeploymentType.BlueGreen,
+        deploymentOption: props.deploymentConfig === LambdaDeploymentConfig.AllAtOnce ?
+          DeploymentOption.WithoutTrafficControl : DeploymentOption.WithTrafficControl
+      },
+
+      alarmConfiguration: new cdk.Token(() => renderAlarmConfiguration(alarms, props.ignorePollAlarmsFailure)),
+      autoRollbackConfiguration: new cdk.Token(() => renderAutoRollbackConfiguration(alarms, props.autoRollback)),
+    });
+
+    this.application = props.application;
+    this.deploymentGroupName = resource.deploymentGroupName;
+    this.deploymentGroupArn = deploymentGroupNameToArn(this.application.applicationName, this.deploymentGroupName, this);
+
+    if (props.preHook) {
+      this.grantPutLifecycleEventHookExecutionStatus(props.preHook.role);
+      props.preHook.grantInvoke(serviceRole);
+    }
+    if (props.postHook) {
+      this.grantPutLifecycleEventHookExecutionStatus(props.postHook.role);
+      props.postHook.grantInvoke(serviceRole);
+    }
+
+    (props.alias.node.findChild('Resource') as lambda.CfnAlias).options.updatePolicy = {
+      codeDeployLambdaAliasUpdate: {
+        applicationName: props.application.applicationName,
+        deploymentGroupName: resource.deploymentGroupName,
+        beforeAllowTrafficHook: props.preHook === undefined ? undefined : props.preHook.functionName,
+        afterAllowTrafficHook: props.postHook === undefined ? undefined : props.postHook.functionName
+      }
+    };
```

Comment:
How do we want to distinguish between deploying by an `UpdatePolicy` and deploying via `CodePipeline`? E.g. the `ServerDeploymentGroup` has an `addToPipelineStage` method but I don't think `UpdatePolicy` and pipeline can co-exist?

Reply:
sounds like they will conflict. is it possible to simply not add the update policy if the deployment group was added to a pipeline? does that make sense?

Reply:
A Lambda Deployment Group can never be added to a CodePipeline.

Reply:
Well that makes things simpler! What's the reasoning behind that?

Reply:
Lambda Deployment Groups just work fundamentally differently than Server Deployment Groups. There is simply no point in putting a Lambda one in a CodePipeline (it wouldn't do anything).